pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -333,8 +354,22 @@ func (ssr *Snapshotter) TakeDeltaSnapshot() (*snapstore.Snapshot, error) {
 		return nil, fmt.Errorf("failed to compute hash of events: %v", err)
 	}
 	ssr.events = hash.Sum(ssr.events)
+
+	rc := ioutil.NopCloser(bytes.NewReader(ssr.events))
+
+	// if compression is enabled
+	//    then compress the snapshot.
+	if ssr.compressionConfig.Enabled {
+		ssr.logger.Info("starting the Compression of delta snapshot")
+		startCompressTime := time.Now()
+		rc = compressor.CompressSnapshots(rc, ssr.compressionConfig.CompressionPolicy)
+		totalCompressTime := time.Now().Sub(startCompressTime).Seconds()
+		ssr.logger.Infof("Successfully compresses the delta snapshot in %v seconds", totalCompressTime)
```

Comment:
This logging should be done after the `ssr.store.Save()` call where the stream is read -> https://github.com/gardener/etcd-backup-restore/pull/293/files?file-filters%5B%5D=.go#diff-5c59e1a2063c6043cf7cad27aee199b833950c80ec75202ac69699216210a2f1R372

Reply:
same Counter-argument as above

Reply:
Please see https://github.com/gardener/etcd-backup-restore/pull/293#discussion_r538283599